service/history/queueProcessor.go

```diff
@@ -343,9 +354,11 @@ func redispatchQueueTasks(
 	redispatchQueue collection.Queue,
 	queueTaskProcessor queueTaskProcessor,
 	logger log.Logger,
+	metricsScope metrics.Scope,
 	shutdownCh <-chan struct{},
 ) {
 	queueLength := redispatchQueue.Len()
+	metricsScope.RecordTimer(metrics.TaskRedispatchQueuePendingTasksTimer, time.Duration(queueLength))
```

Comment:
consider using a per-shard gauge metric instead of a timer. Per shard metrics are ok as long as no rollups are needed at visualization time (which we won't need in this case).